pkg/resource/role/hooks.go

```diff
@@ -342,6 +342,36 @@ func (rm *resourceManager) removeInlinePolicy(
 	return err
 }
 
+// syncAssumeRolePolicies syncs the role's assume role policy document.
+func (rm *resourceManager) syncAssumeRolePolicies(
+	ctx context.Context,
+	r *resource,
+) (err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.syncAssumeRolePolicies")
+	defer func() { exit(err) }()
+	return rm.putAssumeRolePolicies(ctx, r)
```

Comment:
Similar calls to `sync$OBJECT` use a `delete` variant if the `spec` is empty. For example

https://github.com/aws-controllers-k8s/iam-controller/blob/3bac44622ed2e9c5420e092b6cc51081957b831d/pkg/resource/role/hooks.go#L73-L75


In the case of the assume policy document, I could not see the AWS API call equivalent to "DeleteAssumeRoleDocument", instead it seems that simply "putting" the blank document is sufficient.

I was weary of adding in duplicate code for the call, with a simple rename to `deleteAssumeRolePolicies` as I felt this may be quite confusing so I've left it out. If there is a belief that this should be done, I can add it in to keep the consistency with other functions 😊